### PR TITLE
Add Lint CI

### PR DESCRIPTION
**Pull request by zunda-pixel**
## Motivation

I know Auto Formatter Action is already set.
But Sometimes formatter(swift-format) breaks source codes, Then manual format is better.

Committer can find format issue at creating pull request.

- Add swift-format package to MyLibrary
- Add `Lint CI on pull requests`
- Add `.DS_Store` to `.gitignore`

Developer can format code with Xcode Plugin before create pull requests.

Xcode Plugin

### CODE REVIEW

**Comment by d-date** [comment]
Hi @zunda-pixel ,

Thank you for trying to contribution.  Let me know your motivation about this PR.
The formatter behavior might be different from apple/swift-format.

FYI: Now formatting on CI is fixed.

**Comment by zunda-pixel** [comment]
I added Motivation.
But Resolving lint issue is little hard.
if lint is bothersome or lint ci system has some problems, Please close this pr.

File: MyLibrary/Sources/SponsorFeature/Sponsors.swift [hunk 1]
```diff
@@ -43,7 +43,7 @@ public struct SponsorsList {
         state.sponsors = try! dataClient.fetchSponsors()
         return .none
 
-      case let .view(.sponsorTapped(sponsor)):
+      case .view(.sponsorTapped(let sponsor)):
```
Comment (d-date) [inline MyLibrary/Sources/SponsorFeature/Sponsors.swift:46]: What rule of swift-fomat is applied about precedence of `let` keyword?

Comment (zunda-pixel) [reply MyLibrary/Sources/SponsorFeature/Sponsors.swift:46]: this rule is [UseLetInEveryBoundCaseVariable](https://github.com/apple/swift-format/blob/main/Documentation/RuleDocumentation.md#useletineveryboundcasevariable)
swift-format doesn't format this rule.

**Pull request closed without merging**